models/staking/ethereum/staking_ethereum_deposits.sql

```diff
@@ -13,56 +14,56 @@
 WITH deposit_events AS (
     SELECT d.evt_block_time AS block_time
     , d.evt_block_number AS block_number
-    , CAST(d.evt_index AS VARCHAR(100)) AS evt_index
+    , d.evt_index
     , d.evt_tx_hash AS tx_hash
-    , CAST(bytea2numeric_v3('0x' || SUBSTR(d.amount, 15, 2) || SUBSTR(d.amount, 13, 2) || SUBSTR(d.amount, 11, 2) || SUBSTR(d.amount, 9, 2) ||
-        SUBSTR(d.amount, 7, 2) || SUBSTR(d.amount, 5, 2) || SUBSTR(d.amount, 3, 2))/POWER(10, 9) AS DOUBLE) AS amount
+    , from_big_endian_64(reverse(d.amount)) / POWER(10, 9) AS amount
     , d.contract_address
-    , CAST(bytea2numeric_v3('0x' || SUBSTR(d.index, 15, 2) || SUBSTR(d.index, 13, 2) || SUBSTR(d.index, 11, 2) || SUBSTR(d.index, 9, 2) ||
-        SUBSTR(d.index, 7, 2) || SUBSTR(d.index, 5, 2) || SUBSTR(d.index, 3, 2)) AS DECIMAL(38,0)) AS deposit_index
+    , from_big_endian_64(reverse(d.index)) AS deposit_index
     , d.pubkey
     , d.signature
-    , CAST(substring(d.withdrawal_credentials, 1, 4) AS string) AS withdrawal_credentials_type
-    , CASE WHEN substring(d.withdrawal_credentials, 1, 4) = '0x01' THEN CAST('0x' || substring(d.withdrawal_credentials, -40, 40) AS string) ELSE CAST(NULL AS string) END AS withdrawal_address
+    , substring(CAST(d.withdrawal_credentials AS varchar), 1, 4) AS withdrawal_credentials_type
+    , CASE WHEN substring(CAST(d.withdrawal_credentials AS varchar), 1, 4) = '0x01' THEN substr(d.withdrawal_credentials, -20)
```

Comment:
for these, should we also avoid cast to varchar, but rather use [this](https://dune.com/docs/query/DuneSQL-reference/Functions-and-operators/varbinary/#bytearray_substring)?

Reply:
Alright, I switched to using `bytearray_position()` & `bytearray_substring()` instead of string casting